app/notebook/server/NotebookManager.scala

```diff
@@ -54,7 +54,9 @@ class NotebookManager(val name: String, val notebookDir: File) {
     val sep = if (path.last == '/') "" else "/"
     val fpath = name.map(path + sep + _ + extension).getOrElse(incrementFileName(path + sep + "Untitled"))
     val nb = Notebook(
-      Some(new Metadata(getName(fpath),
+      Some(Metadata(
```

Comment:
I'd assume this is for backward compatibility, in case some old format didn't contain it.